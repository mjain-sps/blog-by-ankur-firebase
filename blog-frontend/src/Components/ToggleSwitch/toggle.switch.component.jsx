import React from "react";
import { ToggleLabel, ToggleInput, ToggleSpan } from "./toggle.switch.styles";

const ToggleSwitch = ({ id, checked, value, rounded, onChange }) => {
  return (
    <ToggleLabel rounded={rounded} htmlFor={id}>
      <ToggleInput
        type="checkbox"
        id={id}
        checked={checked}
        value={value}
        onChange={onChange}
      />
      <ToggleSpan rounded={rounded} />
    </ToggleLabel>
  );
};

export default ToggleSwitch;
