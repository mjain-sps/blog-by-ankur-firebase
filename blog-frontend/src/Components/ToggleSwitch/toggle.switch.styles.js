import styled, { css } from "styled-components";
import MainTheme from "../../Theme";
const ToggleInput = styled.input`
  opacity: 0;
  width: 0px;
  height: 0px;
  &:checked ~ span {
    transform: translateX(26px);
  }
`;

const ToggleLabel = styled.label`
  width: 60px;
  height: 34px;
  background-color: ${MainTheme.backgroundColour.BG_ORANGE};
  position: relative;
  display: inline-block;
  ${(props) =>
    props.rounded &&
    css`
      border-radius: 26px;
    `}
`;

const ToggleSpan = styled.span`
  position: absolute;
  top: 0%;
  left: 0%;
  right: 0%;
  bottom: 0%;
  cursor: pointer;
  transition: 400ms ease;

  &::before {
    position: absolute;
    content: "";
    height: 26px;
    width: 26px;
    top: 4px;
    left: 4px;
    background-color: white;

    ${(props) =>
      props.rounded &&
      css`
        border-radius: 26px;
      `}
  }
`;

export { ToggleLabel, ToggleInput, ToggleSpan };
