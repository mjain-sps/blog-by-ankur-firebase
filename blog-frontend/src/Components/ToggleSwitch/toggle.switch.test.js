import React from "react";
import { render } from "@testing-library/react";
import ToggleSwitch from "./toggle.switch.component";

test("snapshot testing of component", () => {
  const { container } = render(<ToggleSwitch />);
  expect(container).toMatchSnapshot();
});
