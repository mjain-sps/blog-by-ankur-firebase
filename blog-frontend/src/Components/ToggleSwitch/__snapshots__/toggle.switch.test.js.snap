// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing of component 1`] = `
<div>
  <label
    class="sc-gtsrHT geEOoQ"
  >
    <input
      class="sc-bdnxRM hwrWxp"
      type="checkbox"
      value=""
    />
    <span
      class="sc-dlnjwi cfPyCM"
    />
  </label>
</div>
`;
